data-raw/paper/paper.Rmd

```diff
@@ -0,0 +1,73 @@
+
+# Underlying Principles
+
+At a high level, stacked ensemble models in {stacks} are formed from *model definitions*, which specify a set of instructions to fit a model or set of models using {parsnip}, {recipes}, and {workflows}. Each model specified in a model definition is referred to as a *candidate member* in the package. To be used in the same model stack, all model definitions must share a *resampling object*, as defined in the {rsample} package, such as a cross-fold validation or set of bootstrap samples. After initializing a `data_stack` object with the `stacks()` function, model definitions can be iteratively added to the data stack with `add_candidates()`. This function collates the predictions on the assessment set, a subset of the training data used for preliminary model validation, from each candidate specified in the model definitions to the data stack.
+
+```{r, echo = FALSE, out.width = '100%'}
+knitr::include_graphics("class_data_stack.png")
+```
+
+The above diagram represents a data stack containing 11 candidate members, where 4 come from a shared K-nearest neighbors model definition, 1 arises from a linear model, and 6 come from a shared support vector machine model definition. Model definitions can result in multiple candidates when several possible hyperparameter values are being optimized over (e.g. the number of neighbors $k$ or the precision parameter for the support vector machine's radial basis function).
+
+After all candidates are collated to the data stack, the user can fit the meta-learner using the `blend_predictions()` function, which fits an elastic net model on the data stack, predicting the true outcome using the predictions from each of the candidate members. The coefficients of this model form the *stacking coefficients*, which are the weightings for each of the member models in ultimately predicting the outcome. Candidate members with non-zero stacking coefficients are *members*, and must be fitted on the entire training set with the `fit_members()` function. This function outputs a `model_stack` object, and is ready to predict on new data.
+
+```{r, echo = FALSE, out.width = '100%'}
+knitr::include_graphics("class_model_stack.png")
+```
+
+In addition to the aforementioned core verbs, the package supplies several helper functions to interface more effectively with model stacks. Notably, `collect_parameters()` juxtaposes model parameters with their stacking coefficients, and an `autoplot()` S3 method provides model diagnostic visualizations using {ggplot2}.
```

Comment:
sAfter this paragraph, maybe give a short block of pseudocode to emulate what a script would have.